Week1/homework/app.js

```diff
@@ -3,9 +3,162 @@
+    },
+    a_tale_of_two_cities: {
+      title: 'A Tale of Two Cities',
+      author: 'Charles Dickens',
+      language: 'English',
+      length: 544,
+      info:
+        'After eighteen years as a political prisoner in the Bastille the aging Dr Manette is finally released and reunited with his daughter in England. There two very different men, Charles Darnay, an exiled French aristocrat, and Sydney Carton, a disreputable but brilliant English lawyer, become enmeshed through their love for Lucie Manette. From the tranquil lanes of London, they are all drawn against their will to the vengeful, bloodstained streets of Paris at the height of the Reign of Terror and soon fall under the lethal shadow of La Guillotine.'
+    },
+    the_alchemist: {
+      title: 'The Alchemist',
+      author: 'Paulo Coelho',
+      language: 'English',
+      length: 197,
+      info:
+        'Paulo Coelhos masterpiece tells the mystical story of Santiago, an Andalusian shepherd boy who yearns to travel in search of a worldly treasure.His quest will lead him to riches far different—and far more satisfying—than he ever imagined.Santiagos journey teaches us about the essential wisdom of listening to our hearts, of recognizing opportunity and learning to read the omens strewn along lifes path, and, most importantly, to follow our dreams.'
+    }
+  };
+  const BookCovers = {
+    moby_dick: 'img/moby_dick.jpg',
+    the_adventures_of_huckleberry_finn: 'img/the_adventures_of_huckleberry_finn.jpg',
+    alices_adventures_in_wonderland: 'img/alices_adventures_in_wonderland.jpg',
+    charlie_and_the_chocolate_factory: 'img/charlie_and_the_chocolate_factory.jpg',
+    wuthering_heights: 'img/wuthering_heights.jpg',
+    war_and_peace: 'img/war_and_peace.jpg',
+    anna_karenina: 'img/anna_karenina.jpg',
+    animal_farm: 'img/animal_farm.jpg',
+    a_tale_of_two_cities: 'img/a_tale_of_two_cities.jpg',
+    the_alchemist: 'img/the_alchemist.jpg'
+  };
+
+  function generateBookCards(container) {
+    for (let i = 0; i < bookTitles.length + 1; i++) {
+      const bookCard = document.createElement('li');
+      bookCard.setAttribute('id', bookTitles[i]);
+      container.appendChild(bookCard);
+    }
+  }
+
+  function insertBookCovers() {
+    const bookList = Object.keys(BookCovers);
+    for (const key of bookList) {
+      const img = document.createElement('img');
+      img.src = BookCovers[key];
+      img.alt = key;
+      const li = document.getElementById(key);
+      li.appendChild(img);
+    }
+  }
+
+  function getActualBookInfo() {
```

Comment:
Usually, if one sees a function named like `get....`, one expects the function to return something at the end. However, in this function, we aren't returning anything. Instead, we are inserting all the book related information to our list elements. A more appropriate name for this function could have been `insertBookDetails` or something similar to how you named the above function `insertBookCovers`. Not a big thing but it always helps in understanding your code. 🙂

Reply:
I've fixed all the issues upon your review.

I also implemented another function for repeated "create and append  element tasks" in the code as covered in the  last lesson. Should I push and make a new PR for the updated version of homework week1?